app/editPlan/get.controller.test.js

```diff
@@ -12,12 +23,54 @@ describe('getSentencePlanSummary', () => {
   const res = {
     render: jest.fn(),
   }
+  afterEach(() => {
+    res.render.mockReset()
+    getAboutTheIndividual.mockReset()
+    getAddObjectives.mockReset()
+    getFinalInformation.mockReset()
+  })
 
-  describe('set up rendering', () => {
-    it('should set the correct render values when there is an active plan', async () => {
-      const expected = { id: 1, token: '1234' }
-      await editPlan(req, res)
-      expect(res.render).toHaveBeenCalledWith(`${__dirname}/index`, expected)
+  describe('editPlan', () => {
+    const completionStatus = true
+    const aboutTheIndividualSection = { value: 'section 1' }
+    const addObjectivesSection = { value: 'section 2', items: [{ complete: completionStatus }] }
+    const finalInformationSection = { value: 'section 3' }
+    const planSummary = { sections: [aboutTheIndividualSection, addObjectivesSection, finalInformationSection] }
+    beforeEach(() => {
+      getAboutTheIndividual.mockReturnValueOnce(aboutTheIndividualSection)
+      getAddObjectives.mockReturnValueOnce(addObjectivesSection)
+      getFinalInformation.mockReturnValueOnce(finalInformationSection)
+    })
+    describe('with a valid sentence plan', () => {
+      beforeEach(async () => {
+        getSentencePlan.mockReturnValueOnce(sentencePlan)
+        await editPlan(req, res)
+      })
+      it('should generate the "About the individual" section', () => {
+        expect(getAboutTheIndividual).toHaveBeenCalledWith(sentencePlan, path)
+      })
+      it('should generate the "Add objectives" section', () => {
+        expect(getAddObjectives).toHaveBeenCalledWith(sentencePlan, path)
+      })
+      it('should generate the "Final information" section', () => {
+        expect(getFinalInformation).toHaveBeenCalledWith(sentencePlan, path, completionStatus)
+      })
+      it('should set the correct render value', () => {
+        const expected = { id: 1, token: '1234', planSummary }
+        expect(res.render).toHaveBeenCalledWith(`${__dirname}/index`, expected)
+      })
+    })
+    describe('on recieving an error', () => {
```

Comment:
receiving

(sorry, I can't leave it!)

Reply:
Hang on, I'll just check the rule:
_**i** before **e** except after **c**_
No, it looks like you are correct 👍